gotham/src/middleware/session/mod.rs

```diff
@@ -491,7 +541,7 @@ where
         cookie_config: SessionCookieConfig,
     ) -> NewSessionMiddleware<B, T> {
         NewSessionMiddleware {
-            cookie_config: Arc::new(cookie_config),
+            cookie_config: Arc::new(cookie_config.validate_prefix()),
```

Comment:
Revised - validating the prefix when `NewSessionMiddleware` is built rather than on cookie serialisation.